### DISCUSSION TITLE
How to install Tenacity?

### DISCUSSION THREAD

**Discussion by TheYellowArchitect** [tenacityteam/tenacity-legacy#581] [Q&A] [answered]
I am a newbie on Linux. I used to use windows, and audacity was easy to install (just download and .exe)

Here, I have no idea. Do I have to build it from source? (never done that)
Any help is appreciated, since I don't want to download latest audacity build here in Linux, since I focus on privacy (even on windows I had, I had an old version of audacity and never updated)

**Comment by fitojb** [comment]
If I were you, I’d wait for the first release to be done. Once a release is created, people can create software packages.

**Reply by poyribn** [reply]
Normally, installing programs on linux Is almost as easy as it is on windows.
It's normally done with a terminal command that is quickly learnt.

Only in special cases like pre-releases (such as with Tenacity right now) is it more complicated than that.
I have used linux for a few months and still can't compile anything from source, even with instructions.
But that's not a problem for me because I have never really needed to do that.

Until Tenacity releases I will still use the old version of audacity (Debian Linux hasn't updated Audacity since version 2, so it doesn't have Muse Group's privacy-breaking changes).
But if in doubt, you can run Audacity in a sandbox without internet, like with the command `firejail net=none audacity`.

**Comment by nbsp** [comment]
If you're running Arch Linux, the `tenacity-git` AUR package can be installed.

If not, you'll have to build Tenacity from source. Follow the instructions in [`BUILDING.md`](https://git.sr.ht/~tenacity/tenacity/tree/master/item/BUILDING.md), and if you run into any problems, join [the IRC channel](https://web.libera.chat/gamja/#tenacity) or shoot an email to [the mailing list](mailto:~tenacity/tenacity-discuss@lists.sr.ht).

Note that this Discussions tab is largely unattended, and we're more active on email or IRC.

**Comment by rubyFeedback** [comment]
> Do I have to build it from source? (never done that)

If you n ever compiled from source then I first recommend
you to compile smaller packages. Start with a hello world
example in C and C++. (Copy / paste from the internet).

When that works, compile somewhat larger projects but
still fairly small-ish.

For instance, libpng is fairly simple to compile:

https://sourceforge.net/projects/libpng/files/libpng16/1.6.37/libpng-1.6.37.tar.xz

Also store info in a local file. It will be helpful to you in the long run, in particular
if you run into problems. Always make sure your toolchain works fine as-is;
distributions may require some dev packages for that.

Sooner or later compiling is no issue. (It can even be done on windows via
msys2 but this is definitely more complicated than on linux; linux makes this
really really easy. Some distributions are more annoying than others, but
by and large linux works VERY well, and even on distributions it works fine.
You may have to first obtain necessary packages. For tenacity itself you
may need some requirements - I think it is on the wiki somewhere or
something, the instructions for that. But I recommend you first start with
smaller things and see that it all works well.

Also check out the LFS / BLFS project for specific programs you want to
compile. Example for libpng:


http://www.linuxfromscratch.org/blfs/view/svn/general/libpng.html

It contains TONS of information. You don't have to install LFS / BLFS;
I use the webpages to get info about problematic programs that are
hard to compile. Most programs compile without problem though!

**Comment by n0toose** [comment, marked as answer]
https://github.com/tenacityteam/tenacity/issues/393
